sources/Outputs/src/Diagram.cpp

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  Diagram.cpp
+ *
+ * @brief Dynaflow launcher Diagram file writer implementation file
+ *
+ */
+
+#include "Diagram.h"
+
+#include "Constants.h"
+
+#include <fstream>
+
+namespace dfl {
+namespace outputs {
+
+Diagram::Diagram(DiagramDefinition&& def) : def_{std::forward<DiagramDefinition>(def)} {}
+
+void
+Diagram::write() {
+  //We don't want to create the file if there are no generators diagrams to write
+  //maybe use stdbuf for using << and not .append(..()))
+  std::stringstream buffer;
+  buffer << "#1";
+  bool empty = true;
+
+  for (algo::GeneratorDefinition& generator : def_.generators) {
+    if (generator.model != algo::GeneratorDefinition::ModelType::DIAGRAM_PQ_SIGNALN and
+        generator.model != algo::GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN)
+      continue;
+    empty = false;
+
+    std::sort(generator.points.begin(), generator.points.end(),
+              [](const DYN::GeneratorInterface::ReactiveCurvePoint& gen1, const DYN::GeneratorInterface::ReactiveCurvePoint& gen2) -> bool {
+                return gen1.p < gen2.p;
+              });
+
+    write_table(generator, buffer, true);
```

Comment:
Please respect the coding style (writeTable)

Reply:
Also, please declare const qmin = true, const qmax = false and use them in the calls to improve lisibility

Reply:
or an enum instead of a boolean

Reply:
Shoud I also change the variables names like is_table_qmin to follow the naming convention ?

Reply:
Oh didn't read far enough, I had not seen your comment farther down